client/src/pages/SocketioConnection.js

```diff
@@ -1,20 +1,27 @@
 import React, { useState, useEffect } from 'react';
 import io from 'socket.io-client';
 let endPoint = "http://127.0.0.1:5000/";
-let socket = io.connect(`${endPoint}`);
+let socket = io.connect(null, {port:5000, rememberTransport: false});
 
 
 function Socketio() {
-  const [messages, setMessages] = useState(["Hello And Welcome"]);
+  const [messages, setMessages] = useState("");
   const [message, setMessage] = useState("");
-
+  const urlElement = window.location.href.split('/')
+  const room_id = urlElement[4]
+  const username = urlElement[5]
+  socket.emit("join",room_id)
   useEffect(() => {
-    getMessages();
+    getMessages()
```

Comment:
since `messages.length` keeps changing as a dependency, this function `getMessages` gets initiated and creates that listener on line 23 (socket.on("messages")) multiple times depending on the length. this should only get intiated once, so if you change line 16 to empty array this should fix it